ARCHITECTURE.md

```diff
@@ -0,0 +1,152 @@
+![BluepillIcon](doc/img/bluepill_text.png)
+
+![](https://github.com/linkedin/bluepill/workflows/master/badge.svg)
+
+## iOS Simulator Architecture
+
+- App (builds with) -> (Xcode/Bazel) -> `X.app` (packaged into an IPA, an archive of the app or "Bundle")
+- iPhones have ARM processors while Macs have `x86_64` (pre M1 at least)
+- Compilation results in an architecture specific binary (or app binary)
+
+## Bluepill Build How-To
+- `bluepill.sh` builds Bluepill in CI (It has a sample app)
+- Technically the script builds Bluepill & the sample app, and runs the tests
+- `BPSampleApp.app` is the "bundle", an x86_64 bundle but iPhone needs an ARM executable
+	- "Fat binary" will have both
+- Side note: iOS is a variant of MacOS and Android ships a VM with a Linux based Android kernel
+- Originally Apple Emulators translated ARM -> `x86_64` but that was very slow!
+	- Helped test iOS apps in Mac by giving you an `x86_64` executable
+- Now they compile all frameworks for `x86_64` and give you a sandbox in which you can build a native binary
+	- It's faster to get an exectuable to test this way
+
+
+## Some notes on iOS Simulators
```

Comment:
This section sort of explains how does simulator work in Mac OS.

And tricks to verify it.